src/layer/unitary.rs

```diff
@@ -0,0 +1,468 @@
+use num;
+use af;
+use af::{Array, MatProp, Dim4, DType};
+
+use utils;
+use std::sync::{Arc, Mutex};
+use activations;
+use initializations;
+use params::{Params};
+use layer::Layer;
+
+use num::Complex;
+
+
+pub struct Unitary {
+    pub input_size: usize,
+    pub output_size: usize,
+}
+
+fn h_d(param: Array, ar: Array) -> Array {
```

Comment:
Add some comments for all these functions. The reader needs to know what this is quickly